internal/util/toml-editor.go

```diff
@@ -0,0 +1,77 @@
+package util
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"os/exec"
+)
+
+// TomlEditorOp is the format of the JSON sent to toml-editor
+type TomlEditorOp struct {
+	Op              string `json:"op"`
+	Path            string `json:"path,omitempty"`
+	TableHeaderPath string `json:"table_header_path,omitempty"`
+	DottedPath      string `json:"dotted_path,omitempty"`
+	Value           string `json:"value,omitempty"`
+}
+
+// TomlEditorResponse is the format of the JSON sent from toml-editor
+type TomlEditorResponse struct {
+	Status  string        `json:"status"`
+	Message string        `json:"message"`
+	Results []interface{} `json:"results"`
+}
+
+func TomlEditorIsAvailable() bool {
+	_, err := exec.LookPath("toml-editor")
+	return err == nil
+}
+
+func ExecTomlEditor(ops []TomlEditorOp) (*TomlEditorResponse, error) {
+	cmd := exec.Command("toml-editor", "--path", "pyproject.toml")
+	stdin, err := cmd.StdinPipe()
+	if err != nil {
+		return nil, fmt.Errorf("toml-editor error: %s", err)
+	}
+	stdout, err := cmd.StdoutPipe()
+	if err != nil {
+		return nil, fmt.Errorf("toml-editor error: %s", err)
+	}
+	err = cmd.Start()
+	if err != nil {
+		return nil, fmt.Errorf("toml-editor error: %s", err)
+	}
+
+	encoder := json.NewEncoder(stdin)
+	err = encoder.Encode(ops)
+	if err != nil {
+		return nil, fmt.Errorf("toml-editor error: %s", err)
+	}
+	err = stdin.Close()
+	if err != nil {
+		return nil, fmt.Errorf("toml-editor error: %s", err)
+	}
```

Comment:
This seems like it would be better suited for after we've consumed everything from `stdout`, to avoid obscure buffer-full scenarios